core/src/proof_generation/mod.rs

```diff
@@ -1,6 +1,157 @@
 //! The proof generation worker handles the core of generating single-prover
 //! proofs for wallet updates
+
+use std::sync::Arc;
+
+use circuits::zk_circuits::{
+    valid_commitments::{ValidCommitments, ValidCommitmentsWitness},
+    valid_reblind::{ValidReblind, ValidReblindWitness},
+    valid_settle::{ValidSettleStatement, ValidSettleWitness},
+    valid_wallet_create::{ValidWalletCreateStatement, ValidWalletCreateWitness},
+    valid_wallet_update::{ValidWalletUpdateStatement, ValidWalletUpdateWitness},
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{MAX_BALANCES, MAX_FEES, MAX_ORDERS};
+
+use self::jobs::{ValidCommitmentsBundle, ValidReblindBundle};
 pub mod error;
 pub mod jobs;
 pub mod proof_manager;
 pub mod worker;
+
+// ----------------------------------
+// | Circuit Default Generics Types |
+// ----------------------------------
+
+/// A witness to `VALID WALLET CREATE` with default size parameters attached
+pub type SizedValidWalletCreateWitness =
+    ValidWalletCreateWitness<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+/// A statement for `VALID WALLET CREATE` with default size parameters attached
+pub type SizedValidWalletCreateStatement =
+    ValidWalletCreateStatement<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+
+/// A `VALID WALLET UPDATE` witness with default const generic sizing parameters
+pub type SizedValidWalletUpdateWitness =
+    ValidWalletUpdateWitness<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+/// A `VALID WALLET UPDATE` statement with default const generic sizing parameters
+pub type SizedValidWalletUpdateStatement =
+    ValidWalletUpdateStatement<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+
+/// A `VALID COMMITMENTS` witness with default const generic sizing parameters
+pub type SizedValidCommitmentsWitness = ValidCommitmentsWitness<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+/// `VALID COMMITMENTS` with default state element sizing
+pub type SizedValidCommitments = ValidCommitments<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+
+/// A `VALID REBLIND` circuit with default const generic sizing parameters
+pub type SizedValidReblind = ValidReblind<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+/// A `VALID REBLIND` witness with default const generic sizing parameters
+pub type SizedValidReblindWitness = ValidReblindWitness<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
```

Comment:
Do we not need sized generic types for the statements for `VALID COMMITMENTS` and `VALID REBLIND`?

Reply:
We will, I'll add them when we need them in the tasks, causes unused lint errors for now